docs/jitserver_tuning.md

```diff
@@ -0,0 +1,125 @@
+  ports:
+    - protocol: TCP
+      port: 38400
+      targetPort: 38400
+  sessionAffinity: ClientIP
+  sessionAffinityConfig:
+    clientIP:
+      timeoutSeconds: 86400
+```
+
+## Resilience
+
+If the client JVM does not find a compatible server to connect to, compilations are performed locally, by the client itself. To account for the case where the server is temporarily unavailable (e.g, server crash followed by Kubernetes launching another server instance), from time to time the client retries to connect to a server at the indicated address and port. The retry mechanism uses an exponential back-off where the retry interval is doubled with each unsuccessful attempt.
+
+## Monitoring
+
+There are several ways to inspect the behavior of a JITServer instance, but all are based on the [OpenJ9 verbose logging facility](https://blog.openj9.org/2018/06/07/reading-verbose-jit-logs/). Note that if the name of the verbose log is not specified, the relevant information is printed to stderr.
+When you use the `-XX:+JITServerLogConnections` command line option, the server prints a message to the verbose log every time a new client JVM connects to it or disconnects from it. This is an easy way to determine that the clients are able to reach the server. E.g. of output:
+```
+#JITServer: t= 74232 A new client (clientUID=14692403771747196083) connected. Server allocated a new client session.
+#JITServer: t= 74282 A new client (clientUID=2599593246759846167) connected. Server allocated a new client session.
+#JITServer: t= 86281 Client (clientUID=14692403771747196083) disconnected. Client session deleted
+```
+
+The server has a heart-beat thread that periodically prints to the verbose log information related to the number of clients connected, the number of active compilation threads, the amount of CPU used, the amount of available memory and the number of times the internal server caches have been cleared. This last bit of information is important for diagnosing performance problems. The heart-beat information is enabled with the following option:
+
+    -Xjit:statisticsFrequency=<period-in-ms>
+
+E.g. of output:
+```
+#JITServer: CurrentTime: Aug 06 17:25:15 2021
+#JITServer: Compilation Queue Size: 0
+#JITServer: Number of clients : 2
+#JITServer: Total compilation threads : 63
+#JITServer: Active compilation threads : 2
+#JITServer: Physical memory available: 14299 MB
+#JITServer: CpuLoad 206% (AvgUsage 25%) JvmCpu 113%
+………..
+```
+A value greater than 0 for the `Compilation Queue Size` is a sign that the server is overloaded. Compilation requests that wait in the compilation queue face greater delays and run the risk of exceeding the timeouts for the networking sockets. To avoid this scenario, you can reduce the number of connected clients, use the techniques described in section [Alleviating CPU congestion at the server](#alleviating-CPU-congestion-at-the-server) or increase the number of compilation threads at the server with the following option:

```

Comment:
Suggest 'exceeding the timeouts for the networking sockets' -> 'exceeding network timeouts'.